src/main.cpp

```diff
@@ -185,16 +161,37 @@ main(int argc, char *argv[])
         if (parser.isSet(debugOption))
                 nhlog::enable_debug_log_from_commandline = true;
 
+        QString appNameWithProfile{"nheko"};
+        if (parser.isSet(configName))
+                appNameWithProfile.append(QString{"-%1"}.arg(parser.value(configName)));
+
+        QSharedMemory mem{appNameWithProfile};
+
+        // Try to create the shared memory. If it already exists, we'll exit
+        if (!mem.create(1)) {
+                // attach and detach to try to delete hanging shared memory
+                // this would be caused by a crash on Unix-based systems (e.g. on a segfault)
+                mem.attach();
+                mem.detach();
+                if (!mem.create(1))
+                        exit(0);
+        }
+
         try {
-                nhlog::init(QString("%1/nheko.log")
+                nhlog::init(QString("%1/%2.log")
                               .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
+                              .arg(appNameWithProfile)
                               .toStdString());
         } catch (const spdlog::spdlog_ex &ex) {
                 std::cout << "Log initialization failed: " << ex.what() << std::endl;
                 std::exit(1);
         }
 
         QSettings settings;
```

Comment:
Do we still need this, if we have UserSettings below?

Reply:
Please elaborate.

Reply:
the settings here still looks redundant, if you have UserSettings right below it

Reply:
Hmm, I think that the QSettings is still used in main(), so I'll leave it for now.